docs_src/design/adr/system-management/0006-Metrics-Collection.md

```diff
@@ -0,0 +1,382 @@
+
+All information (keys, values, tags, etc.) is in string format and placed in a JSON array within the message body.  Here are some example representations:
+
+**Example metric message body with a single value**
+{"name":"service-up", "value":"120", "timestamp":"1602168089665570000", "tags":{"service":"coredata","uom":"days","type":"int64"}}
+
+**Example metric message body with multiple values**
+{"name":"api-requests", "value":"24", "timestamp":"1602168089665570001", "tags":{"service":"coredata","uom":"count","type":"int64", "mean":"0.0665", "rate1":"0.111", "rate5":"0.150","rate15":"0.111"}}
+
+!!! Note
+    The key or metric name must be unique when using go-metrics as it requires the metric name to be unique per the registry.  Metrics are considered immutable.
+
+#### REST endpoints
+
+!!! Note
+    Again, these REST endpoints are per service and meant to control which metrics for a service are turned on or off.  The REST endpoints do not provide metrics data.
+
+- Proposed endpoint for the REST endpoint that responds with what metrics the service offers is:  /api/v2/telemetry
+- Body of the GET response would contain a JSON list of metrics (by metric name key) and an indication of which are `on` and which are `off`
+- Body of the PUT request would contain a JSON list of the metrics that are to be turned `on` (others are assumed to be turned `off`)
+- Body of the PATCH request would contain a JSON list of the metrics that are to be turned `on` or `off` - leaving all other metrics unchanged
+
+#### Configuration
+- Configuration, not unlike that provided in core data or any device service, configuration will specify the message bus type and locations where the metrics messages should be sent.
+- In fact, the message bus configuration will use (or reuse if the service is already using the message bus) the common message bus configuration as defined below.
+- Common configuration for each service for message queue configuration - inclusive of metrics:
+
+``` yaml
+[MessageQueue]
+Protocol = 'redis'  ## or 'tcp'
+Host = 'localhost'
+Port = 5573
+Type = 'redis'  ## or 'mqtt'
+PublishTopicPrefix = "edgex/events/core" # standard and existing core or device topic for publishing  
+  [MessageQueue.Optional]
+  # Default MQTT Specific options that need to be here to enable environment variable overrides of them
+  # Client Identifiers
+  ClientId = "device-virtual"
+  # Connection information
+  Qos = "0" # Quality of Service values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+  KeepAlive = "10" # Seconds (must be 2 or greater)
+  Retained = "false"
+  AutoReconnect = "true"
+  ConnectTimeout = "5" # Seconds
+  SkipCertVerify = "false" # Only used if Cert/Key file or Cert/Key PEMblock are specified
+```
+
+Additional configuration must be provided (in the service configuration.toml) to 

```

Comment:
There's no mention of scheduling other then the presence of `Interval` below.

Reply:
@tonyespy.  the `Interval` is the scheduling.